dumpling/export/util.go

```diff
@@ -78,3 +78,36 @@ func string2Map(a, b []string) map[string]string {
 func needRepeatableRead(serverType version.ServerType, consistency string) bool {
 	return consistency != ConsistencyTypeSnapshot || serverType != version.ServerTypeTiDB
 }
+
+func infiniteChan[T any]() (chan<- T, <-chan T) {
+	in, out := make(chan T), make(chan T)
+
+	go func() {
+		var q []T
+		for {
+			e, ok := <-in
+			if !ok {
+				close(out)
+				return
+			}
+			q = append(q, e)
+			for len(q) > 0 {
+				select {
+				case out <- q[0]:
+					q = q[1:]
+				case e, ok := <-in:
+					if ok {
+						q = append(q, e)
+						break
```

Comment:
if we received a task from `in`, we can't send it to out until we receive another new task. It looks a bit weird. Why don't we simply use one for loop and select in and out channel?

Reply:
got it, I can refine it